server/sonar-web/src/main/js/components/SourceViewer/SourceViewerBase.js

```diff
@@ -248,6 +250,8 @@ export default class SourceViewerBase extends React.PureComponent {
       if (this.mounted) {
         if (response.status === 403) {
           this.setState({ component, loading: false, notAccessible: true });
+        } else if (response.status === 404) {
+          this.setState({ component, loading: false, sourceRemoved: true });
```

Comment:
`sourceRemoved` should be reset to `false` when sources exist

otherwise when you're switching between issues using arrow key, if you open a removed file, then you'll see the warning for every other file